### PR TITLE
Changes on FAST Farm case creation pull request

### PR DESCRIPTION
**Pull request by MYMahfouz**
I have used the pull request #56  as a basis and updated it to include the following:

- Windows batch files.
- Accept .dll controllers
- Create Mann turbulent boxes that are used in the simulation.
- Added a floating flag to include HydroDyn and MoorDyn.

I am planning to make more changes to include:
- Fast Farm batch files including an option for OMP.
- Option to add intial condition according to the wind direction and wind speed.
- Adding wave height and Tp to the case creation.

I hope this helps. Thanks for creating the library.

BR,
Youssef

### CODE REVIEW

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 1]
```diff
@@ -182,92 +197,27 @@ def __init__(self,
...
-        if self.TShighBoxFilesCreatedBool or self.LESpath is not None:
-            s += f'  High-resolution domain: \n'
-            s += f'   - ds high: {self.ds_high_les} m\n'
-            s += f'   - dt high: {self.dt_high_les} s\n'
-            s += f'   - Extent of high-res boxes: {self.extent_high} D total\n'
-        else:
-            s += f'High-res boxes not created yet.\n'
-        s += f"\n" 
-
-        return s
```
Comment (ebranlard) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:270]: Why was the `__repr__` function removed here? Can you put it back?

File: pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py [hunk 2]
```diff
@@ -20,100 +20,112 @@ def main():
     # -----------------------------------------------------------------------------
 
     # ----------- Case absolute path
-    path = '/complete/path/of/your/case'
+    path = 'path to case'
```
Comment (ebranlard) [inline pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py:23]: Unfortunately, Examples 2,3 and 4 are not "standalone" at the moment and cannot be run. 

Still, it might be best (for a git history perspective), if you only include in this example the changes relevant to your commit (do not change hub height, frequency, number of turbines, paths, tmax, etc). 

To separate the batch and slurm code, you can add a flag at the top of the script, for instance:
```python
 scheduler='batch      # Which jobs scheduler to use ('batch' or 'slurm')
 #scheduler='slurm'  
```
and then for the places where your changes are relevant to batch:
```python
if scheduler=='batch':
    case.copyTurbineFilesForEachCase()
    case.FF_bat_file()
elif scheduler=='slurm':
    # etc
```

File: pyFAST/fastfarm/MannCaseCreation.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        filename=windfilename
+        
+        batch_file_content = "@echo off\n"        
+        batch_file_content += "start \"\"  cmd /C " + Mannpath +" "+ os.path.join(outputpath, filename)  + " " + str(self.alphaepsilon) + " "+ str(self.Lm) + " "+ str(self.gamma) + ' '+ str(self.seedvalue )+ ' '+ str(nx)+ ' '+ str(ny)+ ' '+ str(nz) + ' '+ str(dx)+ ' '+ str(dy)+ ' '+ str(dz) +' false ' + "\n"
+        batch_file_content += "echo Script execution completed\n"
+
+        with open(os.path.join(outputpath, filename+'.bat'), "w") as batch_file:
+            batch_file.write(batch_file_content)  
+            
+        subprocess.run([os.path.join(outputpath, filename+'.bat')],stdout=subprocess.PIPE)
```
Comment (ebranlard) [inline pyFAST/fastfarm/MannCaseCreation.py:93]: You might want to add a flag if the user want to run the batch or not.

**Review by ebranlard — CHANGES_REQUESTED** [review]
Thanks for doing that and sharing. I've added a couple of comments that it would be great to address. Sorry for the extra work!

**Comment by rthedin** [comment]
Hi Youssef-- Thanks for re-submitting the PR. There are a lot of changes here. I will provide detailed comments, but I will also open a discussion so we can talk about your needs and how incorporate the Mann box without modifying defaults. It is imperative that the changes are backward compatible. Let's iterate more about the changes before we pull everything in. Smaller commits with meaningful commit messages (and separate PRs) make everyone's life easier. For instance, all the formatting changes (like removing the f-strings) can/should be done in a separate commit, ideally a separate PR.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 4]
```diff
@@ -75,17 +78,20 @@ def __init__(self,
...
                  extent_low  = None,
                  ffbin = None,
+                 TurbSimbin = None,
+                 Mannbin=None,
                  mod_wake = 1,
                  yaw_init = None,
                  ADmodel = None,
                  EDmodel = None,
-                 nSeeds = 6,
+                 nSeeds = 1,
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:87]: The default number of seeds follow the IEC 61400 standard. If you need to run only one seed for the Mann boxes, then you can specify `nSeeds=1` without the need to change the defaults

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 5]
```diff
@@ -75,17 +78,20 @@ def __init__(self,
...
                  seedValues = None,
                  LESpath = None,
                  sweepWakeSteering = False,
                  sweepYawMisalignment = False,
                  refTurb_rot = 0,
-                 verbose = 0):
+                 verbose = 0,
+                 floating=0):
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:94]: This is certainly a good option to have!

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 6]
```diff
@@ -96,7 +102,9 @@ def __init__(self,
         wts: dictionary
             Wind farm layout and turbine parameters in dictionary form
         tmax: scalar
-            Max simulation time given in seconds
+            Wind field time given in seconds
+        tsim: integer
+            Multiple of tmax to repeat the wind field
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:107]: This seems to be something that only applied to the Mann box? I'm not sure as I haven't used Mann extensively. Changing the description of `tmax` to "wind field time [...]` is misleading because this also the time used on FAST.Farm.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 7]
```diff
@@ -139,7 +147,9 @@ def __init__(self,
         seedValues: list of int
             Seed value for each seed of requested TurbSim simulations if nSeeds!=6
         LESpath: str or list of strings
-            Full path of the LES data, if driven by LES. If None, the setup will be for TurbSim inflow.
+            Full path of the LES data, if driven by LES. 
+            If TurbSim, the setup will be for TurbSim inflow.
+            If MannBox, the setup will be for MannBox inflow.
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:152]: Okay, I see a need for a different option if now we have TurbSim, Mann, and LES. This needs to be changed.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 8]
```diff
@@ -169,6 +182,8 @@ def __init__(self,
         self.extent_low  = extent_low
         self.extent_high = extent_high
         self.ffbin       = ffbin
+        self.TurbSimbin  = TurbSimbin
+        self.Mannbin     = Mannbin
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:186]: Good point. This is useful to have.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 9]
```diff
@@ -182,92 +197,27 @@ def __init__(self,
         self.verbose     = verbose
         self.mod_wake    = mod_wake
         self.attempt     = 1
+        self.floating    = floating
                                         
                                         
-        if self.verbose>0: print(f'Checking inputs...', end='\r')
+        if self.verbose>0: print('Checking inputs...', end='\r')
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:203]: All formatting changes should be submitted as separated commits in separate PRs.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 1 above)
Comment (rthedin) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:270]: It doesn't seem like you pulled the latest from `dev`.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 10]
```diff
@@ -343,33 +293,30 @@ def _checkInputs(self):
         
         # Set domain extents defaults if needed
         default_extent_low  = [3,6,3,3,2]
-        default_extent_high = 1.2          # total extent, half of that to each side.
+        default_extent_high = 1.2
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:360]: I see no reason to delete this comment.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 11]
```diff
@@ -343,33 +293,30 @@ def _checkInputs(self):
...
-            raise ValueError(f'The array for low-res box extents should be given with positive values')
+            raise ValueError('The array for low-res box extents should be given with positive values')
         if not isinstance(self.extent_high, (float,int)):
-            raise ValueError(f'The extent_high should be a scalar')
+            raise ValueError('The extent_high should be a scalar')
         if self.extent_high<=0:
-            raise ValueError(f'The extent of high boxes should be positive')
-        if self.extent_high<1:
-            raise ValueError(f'The extent of high boxes is not enough to cover the rotor diameter. '\
-                              'The extent high is given as the total extent, and it needs to be greater than 1.')
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:361]: Why was this deleted?

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 12]
```diff
@@ -537,22 +483,37 @@ def _create_dir_structure(self):
                 for seed in range(self.nSeeds):
                     seedPath = os.path.join(casePath, f'Seed_{seed}')
                     if not os.path.exists(seedPath):  os.makedirs(seedPath)
-                    turbsimPath = os.path.join(seedPath, 'TurbSim')
-                    if not os.path.exists(turbsimPath):  os.makedirs(turbsimPath)
+                    if self.inflowStr == 'MannBox':
+                        path=os.path.dirname(self.path)         
+                        WindPath = os.path.join(path, 'Wind')
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:552]: I'm not sure I'm following your naming convention. Can you explain better what you are trying to do here and how it would fit on workflows already in place using the updated class?

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 1 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:270]: I started working on this before the `__repr__` function was added. I modified this

File: pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py (hunk 2 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py]: Done

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 13]
```diff
@@ -1248,20 +1355,12 @@ def TS_low_slurm_prepare(self, slurmfilepath):
...
-        options = f"--qos='{qos}' "
-        if A is not None:
-            options += f'-A {A} '
-        if t is not None:
-            options += f'-t {t} '
-
-        sub_command = f"sbatch {options}{self.slurmfilename_low}"
-        print(f'Calling: {sub_command}')
-        _ = subprocess.call(sub_command, cwd=self.path, shell=True)
+        _ = subprocess.call(f'sbatch {self.slurmfilename_low}', cwd=self.path, shell=True)
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:1428]: Unclear why you would remove extra options.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 14]
```diff
@@ -1301,15 +1428,15 @@ def getDomainParameters(self):
             raise ValueError(f'The number of cases do not match as expected. {self.nHighBoxCases} unique wind directions, but {len(self.allHighBoxCases.case)} unique cases.')
         
         # Determine offsets from turbines coordinate frame to TurbSim coordinate frame
-        self.yoffset_turbsOrigin2TSOrigin = -( (self.TSlowbox.ymax - self.TSlowbox.ymin)/2 + self.TSlowbox.ymin )
+        self.yoffset_turbsOrigin2TSOrigin =np.round( -( (self.TSlowbox.ymax - self.TSlowbox.ymin)/2 + self.TSlowbox.ymin ),4)
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:1495]: This is likely a good addition. Thank you!

**Comment by MYMahfouz** [comment]
Hi @rthedin ,
I am sorry I wasn't really keeping track of the changes. I did my best not to do any big changes to the code that was already there.

I believe there are two major changes:
- The way I rotate the turbines in the `_rotate_wts`.
- I stopped using the symlink as it was causing some errors and I copied all the Low.bts file instead. This is however not a problem using the MannBox as only one wind field is used.

I am sorry for the mess and the big changes, I am willing to help and adjust as needed. However, I cannot redo everything step by step and recommit as this will be very time consuming. 
Again thanks a lot for creating this code as it is super helpful.

Just one question. I never managed to run FASTFarm with `Mod_AmbWind=3`, did you run it successfully?

BR,
Youssef

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 15]
```diff
@@ -1776,43 +1936,46 @@ def _FF_setup_TS(self):
...
+                    # if self.mod_wake == 1: # Polar model
+                    #     self.dr = self.cmax
+                    # else: # Curled; Cartesian
+                    #     self.dr = round(self.D/10)
                     ff_file['dr'] = self.dr
                     ff_file['NumRadii']  = int(np.ceil(3*D_/(2*self.dr) + 1))
                     #ff_file['NumPlanes'] = int(np.ceil( 20*D_/(dt_low_desired*Vhub_*(1-1/6)) ) )
-                    ff_file['NumPlanes'] = int(np.ceil( 20*D_/(self.dt_low_les*Vhub_*(1-1/6)) ) )
+                    # ff_file['NumPlanes'] = int(np.ceil( self.extent_low[1]*D_/(self.dt_low_les*Vhub_*(1-1/6)) ) )
+                    ff_file['NumPlanes'] = int(np.ceil( self.extent_low[1]*D_/(self.dt_low_les*Vhub_ ) ))
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:2036]: This is not what is on the modeling guidance.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 16]
```diff
@@ -2022,15 +2194,397 @@ def FF_slurm_submit(self, qos='normal', A=None, t=None, delay=4):
...
+                        options += f'-t {t}'
 
-                    sub_command = f"sbatch {options}{fname}"
+                    sub_command = f"sbatch {options} {fname}"
                     print(f'Calling: {sub_command}')
                     _ = subprocess.call(sub_command, cwd=self.path, shell=True)
                     time.sleep(delay) # Sometimes the same job gets submitted twice. This gets around it.
 
 
+    def Create_Mannbox(self, Lm, gamma):
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py]: Thanks for creating this function. It is a really good capability to add to this case generation tool.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 17]
```diff
@@ -2022,15 +2194,397 @@ def FF_slurm_submit(self, qos='normal', A=None, t=None, delay=4):
...
+                        if dx == None:
+                            dx=re.search('Dx : ((?:0|[1-9]\d*)(\.\d*)?(?:[eE][+\-]?\d+))', lines[i])
+                        if ny == None:
+                            ny=re.search('Ny :(\d+)', lines[i])
+                        if dy == None:
+                            dy=re.search('Dy : ((?:0|[1-9]\d*)(\.\d*)?(?:[eE][+\-]?\d+))', lines[i])
+                        if nz == None:
+                            nz=re.search('Nz :(\d+)', lines[i])
+                        if dz == None:
+                            dz=re.search('Dz : ((?:0|[1-9]\d*)(\.\d*)?(?:[eE][+\-]?\d+))', lines[i])
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:2370]: Regex like this should come with some comments above. It is hard to tell what is being done here.

File: pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py [hunk 18]
```diff
@@ -20,100 +20,112 @@ def main():
...
+    turbsimLowfilepath      = '\SampleFiles\template_Low_InflowXX_SeedY.inp'
+    turbsimHighfilepath     = '\SampleFiles\template_HighT1_InflowXX_SeedY.inp'
     
     # SLURM scripts
-    slurm_TS_high           = './SampleFiles/runAllHighBox.sh'
-    slurm_TS_low            = './SampleFiles/runAllLowBox.sh'
-    slurm_FF_single         = './SampleFiles/runFASTFarm_cond0_case0_seed0.sh'
+    slurm_TS_high           = '\SampleFiles/runAllHighBox.sh'
+    slurm_TS_low            = '/runAllLowBox.sh'
+    slurm_FF_single         = '/runFASTFarm_cond0_case0_seed0.sh'
```
Comment (rthedin) [inline pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py]: I see how I left the `/` here for *nix systems and should have used `os.path.join` instead. But by changing this to the Windows convention, you break its use. This needs to be updated to a more generic approach (e.g. using `os`)

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 19]
```diff
@@ -623,79 +620,127 @@ def copyTurbineFilesForEachCase(self, writeFiles=True):
...
                         self.SElastoDynFile['RotSpeed'] = self.bins.sel(wspd=Vhub_, method='nearest').RotSpeed.values
                         self.SElastoDynFile['NacYaw']   = yaw_deg_ + yaw_mis_deg_
                         if writeFiles:
                             self.SElastoDynFile.write(os.path.join(currPath,f'{self.SEDfilename}{t+1}_mod.dat'))
-        
+                            
                     # Update each turbine's ServoDyn
                     self.ServoDynFile['YawNeut']      = yaw_deg_ + yaw_mis_deg_
-                    self.ServoDynFile['DLL_FileName'] = f'"{self.DLLfilepath}{t+1}.so"'
+                    self.ServoDynFile['DLL_FileName'] = f'"{self.DLLfilepath}{t+1}.dll"'
```
Comment (rthedin) [inline pyFAST/fastfarm/FASTFarmCaseCreation.py:722]: `so` are used in *nix and `dll` in Windows. We need to identify what system the user is using or not specify the extension.

**Review by rthedin — COMMENTED** [review]
Apologies for the amount of comments. Let's move this to a discussion and iterate how to properly integrate the Mann boxes and your workflow into the toolbox.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 6 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:107]: This is also an option in TurbSim. Your simulation time can be multiple of your windfield time, this way you do not create big windfields.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 4 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:87]: agreed I was just making it easier for myself but you are correct.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 10 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:360]: For floating this is not a big enough default and might be misleading for whoever is running the simulation. I prefer if we give an error than to set a default value.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 20]
```diff
@@ -568,45 +529,81 @@ def copyTurbineFilesForEachCase(self, writeFiles=True):
...
-                    self.HydroDynFile['WvLowCOffS'] = 2.0*np.pi/self.HydroDynFile['WaveTp']
-                    if writeFiles:
-                        self.HydroDynFile.write(os.path.join(currPath, self.HDfilename))
+                # Update parameters to be changed in the *Dyn files
+                self.HydroDynFile['WaveMod']     = 2
+                # self.HydroDynFile['WaveHs']     = self.bins.sel(wspd=Vhub_, method='nearest').WaveHs.values
+                # self.HydroDynFile['WaveTp']     = self.bins.sel(wspd=Vhub_, method='nearest').WaveTp.values
+                
+                self.HydroDynFile['WaveHs']     = 2
+                self.HydroDynFile['WaveTp']     = 6
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:602]: I agree i just used an easy option for now

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 21]
```diff
@@ -568,45 +529,81 @@ def copyTurbineFilesForEachCase(self, writeFiles=True):
                 # Recover info about the current CondXX_*/CaseYY_*
                 Vhub_ = self.allCond.sel(cond=cond)['vhub'].values
         
-                # Update parameters to be changed in the HydroDyn files
-                if self.HydroDynFile != 'unused':
-                    self.HydroDynFile['WaveHs']     = self.bins.sel(wspd=Vhub_, method='nearest').WaveHs.values
-                    self.HydroDynFile['WaveTp']     = self.bins.sel(wspd=Vhub_, method='nearest').WaveTp.values
-                    self.HydroDynFile['WvHiCOffD']  = 2.0*np.pi/self.HydroDynFile['WaveTp']
-                    self.HydroDynFile['WvLowCOffS'] = 2.0*np.pi/self.HydroDynFile['WaveTp']
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:576]: I believe the wave height and wave period should be an input like wind speed that goes into the case creation. 
The other option would be to allow the user to give a csv file as an input. This csv can also have initial conditions.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 22]
```diff
@@ -623,79 +620,127 @@ def copyTurbineFilesForEachCase(self, writeFiles=True):
...
+                                self.HydroDynFile['PotFile'] = os.path.join(self.templatePath,'HydroData', f'Activefloat_T{t+1}')
+                                self.HydroDynFile.write(os.path.join(currPath, f'{self.HDfilename}{t+1}_mod.dat'))
+                                self.MoorDynFile     = self._check_and_open(self.MDfilepath+f'{t+1}.dat')
+                                for line in range(len(self.MoorDynFile['ConnectionProp'])):
+                                    x1=float(self.MoorDynFile['ConnectionProp'][line][2])
+                                    y1=float(self.MoorDynFile['ConnectionProp'][line][3])
+                                    x_rot = x1*cosd(self.inflow_deg[case]) + y1*sind(self.inflow_deg[case])    
+                                    y_rot = -x1*sind(self.inflow_deg[case]) + y1*cosd(self.inflow_deg[case])
+                                    self.MoorDynFile['ConnectionProp'][line][2]= str(round(x_rot,2))
+                                    self.MoorDynFile['ConnectionProp'][line][3]= str(round(y_rot,2))
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:709]: I am rotating the mooring lines. Since the wind direction cannot be changed and we are rotating the layout. The mooring anchors and fairleads should also be rotated. This is what I am doing here.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 23]
```diff
@@ -1118,27 +1195,48 @@ def _create_all_cases(self):
 
 
     def _rotate_wts(self):
+        '''
+        The function rotates the wind farm according to the wind direction
+        after rotation the wind farm is pushed back to make sure the X0_ low is equal to zero.
+        This makes us sure that this will still work for Mann turbulent boxes for all wind directions.
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:1265]: Yes

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 13 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:1428]: Sorry I was trying to run this on a linux machine and was playing with the options

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 24]
```diff
@@ -1281,8 +1380,36 @@ def TS_low_createSymlinks(self):
         os.chdir(notepath)
 
 
-    def getDomainParameters(self):
+    def TS_low_bat_file(self):
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:1447]: I am writing it from scratch

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 15 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:2036]: No it is the same. This just fits the low and high resolution grids. This assures that the origin of the high resolution grid is a node in the low resolution grid.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 25]
```diff
@@ -1331,16 +1458,19 @@ def TS_high_get_time_series(self):
...
                         HubHt_   = self.allCases.sel(case=case, turbine=t)['zhub'].values
                         xloc_    = self.allCases.sel(case=case, turbine=t)['Tx'  ].values
                         yloc_    = self.allCases.sel(case=case, turbine=t)['Ty'  ].values
         
                         # Turbine location in TurbSim reference frame
-                        xt = xloc_ + self.xoffset_turbsOrigin2TSOrigin
-                        yt = yloc_ + self.yoffset_turbsOrigin2TSOrigin
+                        # xt = xloc_ + self.xoffset_turbsOrigin2TSOrigin
+                        xt = xloc_
+                        yt = np.round(yloc_ + self.yoffset_turbsOrigin2TSOrigin,4)
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:1537]: You are right I am sorry I couldn't make TurbSim work and was trying to figure out why.

File: pyFAST/fastfarm/FASTFarmCaseCreation.py [hunk 26]
```diff
@@ -1776,43 +1936,46 @@ def _FF_setup_TS(self):
...
                     # Wake dynamics
                     ff_file['Mod_Wake'] = self.mod_wake
-                    if self.mod_wake == 1: # Polar model
-                        self.dr = self.cmax
-                    else: # Curled; Cartesian
-                        self.dr = round(self.D/10)
+                    # if self.mod_wake == 1: # Polar model
+                    #     self.dr = self.cmax
+                    # else: # Curled; Cartesian
+                    #     self.dr = round(self.D/10)
```
[the remark this answers is not in the document]
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:2031]: yes. The dr should be same as DX and this is not necessarily the same as self.cmax

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 17 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:2370]: I am reading the parameters of the MannBox as they have to be identical to the parameters in the Inflowind.dat file

File: pyFAST/fastfarm/FASTFarmCaseCreation.py (hunk 19 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/FASTFarmCaseCreation.py:722]: yes I agree

File: pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py (hunk 18 above)
Comment (MYMahfouz) [reply pyFAST/fastfarm/examples/Ex3_FFarmCompleteSetup.py]: yes I agree

**Comment by MYMahfouz** [comment]
@rthedin no worries. I have replied to all the comments.

BR,
Youssef

**Comment by rthedin** [comment]
Closing this and moving it to a discussion: https://github.com/OpenFAST/python-toolbox/discussions/67.
